### PR TITLE
ft: ZENKO-1240 ingestion bucket search in mongo

### PR DESCRIPTION
**Pull request by philipyoo**
Add the ability to find ingestion buckets directly from mongo.

### CODE REVIEW

**Comment by philipyoo** [comment]
@bert-e wait

File: lib/storage/metadata/mongoclient/MongoClientInterface.js [hunk 1]
```diff
@@ -1366,6 +1366,31 @@ class MongoClientInterface {
...
+        m.find({
+            _id: {
+                $nin: [PENSIEVE, constants.usersBucket],
+            },
+            'value.ingestion': {
+                $exists: true,
+            },
+        }, {
+            'value.name': true,
+            'value.ingestion': true,
```
Comment (JianqinWang) [inline lib/storage/metadata/mongoclient/MongoClientInterface.js:1382]: right now there are 3 possible values for ingestion: `{ Status: 'Enabled' }`, `{ Status: 'Disabled' }`, `null`

**Review by rahulreddy — CHANGES_REQUESTED** [review]
I would not recommend leaking application logic into abstractions. They have to independent. What you can do is you can extend `MongoClientInterface` in Backbeat directly and write your own method there.

File: lib/storage/metadata/mongoclient/MongoClientInterface.js (hunk 1 above)
Comment (philipyoo) [reply lib/storage/metadata/mongoclient/MongoClientInterface.js:1382]: this arg is to indicate the return value should include the BucketInfo `ingestion` field, so unless the value is null, it should return the bucket

**Comment by philipyoo** [comment]
@rahulreddy sounds good, closing and will implement an extension in backbeat

File: lib/storage/metadata/mongoclient/MongoClientInterface.js (hunk 1 above)
Comment (JianqinWang) [reply lib/storage/metadata/mongoclient/MongoClientInterface.js:1382]: my bad, misread

**Pull request closed without merging**